src/solc_0.8/defi/rules/ContributionRules.sol

```diff
@@ -0,0 +1,205 @@
+    event ERC721MultiplierDeleted(address indexed contractERC721);
+
+    modifier _isContract(address account) {
+        require(account.isContract(), "ContributionRules: invalid address");
+
+        _;
+    }
+
+    function computeMultiplier(address account, uint256 amountStaked) external view returns (uint256) {
+        uint256 multiplierERC721 = _multiplierBalanceOfERC721(account);
+        uint256 multiplierERC1155 = _multiplierBalanceOfERC1155(account);
+
+        return amountStaked + ((amountStaked * (multiplierERC721 + multiplierERC1155)) / 100);
+    }
+
+    function setERC1155MultiplierList(
+        address contractERC1155,
+        uint256[] memory ids,
+        uint256[] memory multipliers
+    ) external onlyOwner _isContract(contractERC1155) {
+        require(ids.length > 0, "ContributionRules: invalid array of ids");
+        require(multipliers.length > 0, "ContributionRules: invalid array of multipliers");
+
+        IERC1155 multContract = IERC1155(contractERC1155);
+
+        _listERC1155[multContract].ids = ids;
+        _listERC1155[multContract].multipliers = multipliers;
+        _listERC1155[multContract].balanceOf = false;
+
+        // if it's a new member create a new registry, instead, only update
+        if (_islistERC1155Member(multContract) == false) {
+            _listERC1155Index.push(multContract);
+            _listERC1155[multContract].index = _listERC1155Index.length - 1;
+        }
+
+        emit ERC1155MultiplierAdded(contractERC1155, multipliers, ids);
+    }
+
+    function setERC721MultiplierList(
+        address contractERC721,
+        uint256[] memory multipliers,
+        uint256[] memory ids,
+        bool balanceOf
+    ) external onlyOwner _isContract(contractERC721) {
+        IERC721 multContract = IERC721(contractERC721);
+
+        _listERC721[multContract].multipliers = multipliers;
+        _listERC721[multContract].balanceOf = balanceOf;
+        _listERC721[multContract].ids = ids;
+
+        // if it's a new member create a new registry, instead, only update
+        if (_islistERC721Member(multContract) == false) {
+            _listERC721Index.push(multContract);
+            _listERC721[multContract].index = _listERC721Index.length - 1;
+        }
+
+        ERC721MultiplierAdded(contractERC721, multipliers, ids, balanceOf);
+    }
+
+    function getMultiplierERC721List(address reqContract)
+        external
+        view
+        _isContract(reqContract)
+        returns (MultiplierRule memory)
+    {
+        require(_islistERC721Member(IERC721(reqContract)), "ContributionRules: contract is not in the list");
+        return _listERC721[IERC721(reqContract)];
+    }
+
+    function getMultiplierERC1155List(address reqContract)
+        external
+        view
+        _isContract(reqContract)
+        returns (MultiplierRule memory)
+    {
+        require(_islistERC1155Member(IERC1155(reqContract)), "ContributionRules: contract is not in the list");
```

Comment:
make it a modifier

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/commit/3eebb77ee852e2dd4fb5f5a788b3ed2ac64fb5bc